cluster-controller/src/main/java/io/strimzi/controller/cluster/operator/assembly/KafkaAssemblyOperator.java

```diff
@@ -0,0 +1,250 @@
+/*
+ * Copyright 2017-2018, Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.controller.cluster.operator.assembly;
+
+import io.fabric8.kubernetes.api.model.ConfigMap;
+import io.fabric8.kubernetes.api.model.HasMetadata;
+import io.fabric8.kubernetes.api.model.Service;
+import io.fabric8.kubernetes.api.model.extensions.Deployment;
+import io.fabric8.kubernetes.api.model.extensions.StatefulSet;
+import io.strimzi.controller.cluster.Reconciliation;
+import io.strimzi.controller.cluster.operator.resource.ConfigMapOperator;
+import io.strimzi.controller.cluster.operator.resource.DeploymentOperator;
+import io.strimzi.controller.cluster.operator.resource.KafkaSetOperator;
+import io.strimzi.controller.cluster.operator.resource.PvcOperator;
+import io.strimzi.controller.cluster.operator.resource.ReconcileResult;
+import io.strimzi.controller.cluster.operator.resource.ServiceOperator;
+import io.strimzi.controller.cluster.model.KafkaCluster;
+import io.strimzi.controller.cluster.model.Labels;
+import io.strimzi.controller.cluster.model.Storage;
+import io.strimzi.controller.cluster.model.TopicController;
+import io.strimzi.controller.cluster.model.ZookeeperCluster;
+import io.strimzi.controller.cluster.operator.resource.ZookeeperSetOperator;
+import io.vertx.core.AsyncResult;
+import io.vertx.core.CompositeFuture;
+import io.vertx.core.Future;
+import io.vertx.core.Handler;
+import io.vertx.core.Vertx;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static io.strimzi.controller.cluster.model.TopicController.topicControllerName;
+
+/**
+ * <p>Assembly operator for a "Kafka" assembly, which manages:</p>
+ * <ul>
+ *     <li>A ZooKeeper cluster StatefulSet and related Services</li>
+ *     <li>A Kafka cluster StatefulSet and related Services</li>
+ *     <li>Optionally, a TopicController Deployment</li>
+ * </ul>
+ */
+public class KafkaAssemblyOperator extends AbstractAssemblyOperator {
+    private static final Logger log = LoggerFactory.getLogger(KafkaAssemblyOperator.class.getName());
+    private static final String CLUSTER_TYPE_ZOOKEEPER = "zookeeper";
```

Comment:
`CLUSTER_TYPE_ZOOKEEPER` and `CLUSTER_TYPE_TOPIC_CONTROLLER` seem to be unused variables.